core/src/test/java/org/apache/iceberg/TestBaseIncrementalChangelogScan.java

```diff
@@ -132,6 +131,139 @@ public void testFileDeletes() {
     assertThat(t1.existingDeletes()).as("Must be no deletes").isEmpty();
   }
 
+  @TestTemplate
+  public void testPositionDeletes() {
```

Comment:
nice stuff!
Could we have 2 more test cases?
1. It would be nice to test the existing deletes scenario
2. It would be nice to test when the scan returns result for multiple snapshots

Thanks,
Peter

Reply:
Thank you for the review.
I consolidated `testPositionDeletes`, `testEqualityDeletes` and `testMixOfPositionAndEqualityDeletes` into one test to reduce repetition as well as to test a scan with multiple snapshots. I also added tests for scan tasks with existing deletes.
In the process, I discovered a bug in some edge cases and need to work on a fix before I push an update.

Reply:
@wypoon: We usually try to create different tests for different test cases. So your original approach seems nice. Could we remove the code duplication with parameterized tests, or externalizing the duplicated code to test methods?

Reply:
Please look at the new consolidated test, `testRowDeletes`. I think that at the level at which we can test `BaseIncrementalChangelogScan`, it doesn't matter whether delete files are position delete files or equality delete files (they only matter when we actually apply the deletes, which doesn't happen at this level), so `testPositionDeletes` and `testEqualityDeletes` don't really test different cases.

Reply:
Thanks!
I like your new tests!